libethereum/Account.cpp

```diff
@@ -20,11 +20,67 @@
  */
 
 #include "Account.h"
+#include <test/JsonSpiritHeaders.h>
 #include <libethcore/Common.h>
 using namespace std;
 using namespace dev;
 using namespace dev::eth;
+namespace js = json_spirit;
 
 #pragma GCC diagnostic ignored "-Wunused-variable"
 
 const h256 Account::c_contractConceptionCodeHash;
+
+AccountMap dev::eth::jsonToAccountMap(std::string const& _json, AccountMaskMap* o_mask)
+{
+	auto u256Safe = [](std::string const& s) -> u256 {
+		bigint ret(s);
+		if (ret >= bigint(1) << 256)
+			BOOST_THROW_EXCEPTION(ValueTooLarge() << errinfo_comment("State value is equal or greater than 2**256") );
+		return (u256)ret;
+	};
+
+	std::unordered_map<Address, Account> ret;
+
+	js::mValue val;
+	json_spirit::read_string(_json, val);
+
+	for (auto account: val.get_obj().count("alloc") ? val.get_obj()["alloc"].get_obj() : val.get_obj())
```

Comment:
I changed to line from Gavs original version. When calling this function in CanonBlockChain<Ethash>::createGenesisState(), a string with genesis block description is given, not the map directly, therefore I get the "alloc" object here when its there. Could be done in a cleaner way, having another version of this function.